Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -90,8 +90,10 @@ ICompilationStatusNotificationPublisher notificationPublisher
             _ => throw new ArgumentException($"invalid request type ${request.GetType()}")
           };
           request.Document.SetResult(document);
-        } catch (OperationCanceledException e) {
-          request.Document.SetCanceled(e.CancellationToken);
+        } catch (OperationCanceledException) {
```

Comment:
I believe this adaption was made because the catch-clause inside `DocumentDatabase` doesn't have any effect. After applying my proposed refactoring, this error handling should become obsolete.
For design purposes, I refactored the cancellation so that the handling is as close as possible to the issuer of the cancellation. Therefore, `DocumentDatabase` should remain as the ultimate class to decide how to treat a cancellation request and all other classes should bubble the `OperationCancelationException` to the invoker.

Reply:
I think the root cause you're describing here matches my understanding. However, even after the refactoring you suggested above, this change still seems to be necessary. I'll continue investigating why.

Reply:
The latest version of this PR now leaves `TextDocumentLoader` unchanged.